core/src/main/java/com/spotify/metrics/tags/TagExtractor.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2018 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.metrics.tags;
+
+import java.util.Map;
+
+/**
+ * Extract tags to be added to and enrich Metrics.
+ */
+public interface TagExtractor {
+
+    /**
+     * Adds extracted tags to the supplied map.
```

Comment:
It doesn't appear to add to the supplied map, perhaps change to `Creates a new map with the extracted tags from the supplied map.`?